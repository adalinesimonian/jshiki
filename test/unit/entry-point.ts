import jshiki from '../../src'

describe('Entry point', () => {
  describe('parse()', () => {
    it('should parse valid expressions', () => {
      const expression = jshiki.parse('5')

      expect(typeof expression).toBe('function')
      expect(expression()).toBe(5)
    })

    it('should throw an error for invalid expressions', () => {
      expect(() => jshiki.parse('5+')).toThrowErrorMatchingSnapshot()
    })

    it('should parse expressions using the given options', () => {
      const expression = jshiki.parse('x', { scope: { x: '1' } })

      expect(typeof expression).toBe('function')
      expect(expression()).toBe('1')
    })
  })
})
