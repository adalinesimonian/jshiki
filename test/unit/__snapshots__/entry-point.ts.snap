// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry point parse() should throw an error for invalid expressions 1`] = `"Unexpected token <end>"`;
